### DISCUSSION TITLE
function tracing when building with TF

### DISCUSSION THREAD

**Discussion by cwoolfo1** [danielegrattarola/spektral#403] [Q&A]
When I am building models in keras with GAT layers I am getting the following error:

```
        gat_conv_6_layer_call_and_return_conditional_losses(self, inputs, mask, *, training) missing 1 required argument: inputs.
       
        Call arguments received by layer "gat_conv_6" "                 f"(type GATConv):
          • args=(['tf.Tensor(shape=(None, 4, 2), dtype=float32)', 'tf.Tensor(shape=(None, 4, 4), dtype=float32)'],)
          • kwargs=<class 'inspect._empty'>
```

This error is causing all sorts of issues when I try to build models with the functional api. So I decided to subclass a keras model and build it that way but even then I get the same issue when I try to save my model.

The error occurs when I try to trace my call function using empty inputs (where the batch dimension is None).

Any help on fixing this issue would be greatly appreciated

**Comment by danielegrattarola** [comment]
Sorry for the late reply.

Can you post the code that is causing the crash?
